Update Environment variable precedence docs (#2074)
buildkite/docs: pages/pipelines/environment_variables.md

This is the environment your command runs in 🎉

Finally, if your job's commands make any changes to the environment, those changes will only survive as long as the script is running.

### Environment variables in multiple pipeline uploads

> In the event that you are uploading multiple pipeline definitions, depending on the order of uploads, environment variable precedence behaves differently.

Reviewer: General tightening:

```suggestion
If you upload multiple pipeline definitions, the environment variable precedence behaves differently depending on the order of uploads.
```

> If you are uploading pipeline definitions concurrently the value of the environment variable will match the uploading pipeline definition.

Reviewer: Can you please unpack this one a bit more for me? Is it deterministic?

Author: @mbelton-buildkite apologies for the delay there. Yes, it's deterministic.

If we use this example, where we upload the pipeline definitions concurrently:
```yaml
steps:
  - command: |
      buildkite-agent pipeline upload .buildkite/p1.yml
      buildkite-agent pipeline upload .buildkite/p2.yml
```
**p1.yml**

```yaml
env:
  FOO: "bar"

steps:
  - label: "P1" 
    command: echo "+++ p1 - $$FOO"
``` 

**p2.yml**
```yaml
env:
  FOO: "baz"

steps:
  - command: echo "+++ p2 - $$FOO"
```

The resulting pipeline will look like this:

![image](https://github.com/buildkite/docs/assets/19416785/905e7d5e-eb98-4540-b0cd-62b8b6f0935b)

The value of `FOO` will match that of the declaration of `FOO` in the respective uploading pipeline YAML.

> ,

Reviewer: Typo?

```suggestion

```

> Pipeline uploads performed serially will result in the most recent declaration of an environment variable to be true.

Reviewer: I think:

```suggestion
When you upload pipeline definitions in serial, environment variables have the value from the most recent declaration.
```

Is that right?
